### PR TITLE
Clarify the diagnostic message when a local function is left unused

### LINKED ISSUE #9661: Need custom warning for unused local function
An unused local function needs a warning, but not this one:

``` none
                // (9,13): warning CS0168: The variable 'f' is declared but never used
                //         int f() => 2; // local functions
```

### PR DESCRIPTION
**Pull request by alrz**
Fixes #9661

### CODE REVIEW

**Comment by sharwell** [comment]
@gafter I marked this with **Urgency-Soon**. Since this is changing the warning number (as opposed to just the message) for a diagnostic which already shipped to customers, we should make sure to either include it as soon as reasonably possible to avoid a situation where we have to change the message but leave the identifier. If you don't think this type of change would be a problem if it was introduced in 7.2 instead of 7.1, please feel free to remove the label.

**Comment by alrz** [comment]
This is similar to #15706 and possibly #11281, so I think it'd be an acceptable change. I don't know any other diagnostic with different messages based on the context other than perhaps mentioning a specific keyword as part of it.

**Comment by gafter** [comment]
@alrz Please retarget to the 15.6 branch.

**Comment by sharwell** [comment]
@gafter Thank you for taking a look 👍 

@alrz I have retargeted this for you; no action required on your part at this time

File: src/Compilers/CSharp/Portable/Errors/ErrorCode.cs [hunk 1]
```diff
@@ -1486,6 +1486,7 @@ internal enum ErrorCode
         ERR_DefaultLiteralNotValid = 8312,
         WRN_DefaultInSwitch = 8313,
         ERR_PatternWrongGenericTypeInVersion = 8314,
+        WRN_UnreferencedLocalFunction = 8315,
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Errors/ErrorCode.cs]: After the review is done, let's rebase to latest dev15.6 branch and move this to the newly added section for C# 7.2.

**Comment by cston** [comment]
LGTM

**Comment by jcouv** [comment]
@alrz You're good to go with rebasing and resolving conflicts. Thanks

**Comment by jcouv** [comment]
windows_release_vs-integration_prtest failed on Roslyn.VisualStudio.IntegrationTests.VisualBasic.BasicCodeActions.GenerateMethodInClosedFile ([details](https://ci.dot.net/job/dotnet_roslyn/job/dev15.6/job/windows_release_vs-integration_prtest/380/))

FYI @dotnet/roslyn-infrastructure 
I'll re-run

**Comment by jcouv** [comment]
test windows_release_vs-integration_prtest please

**Comment by jcouv** [comment]
Merged. Thanks!

**Pull request merged**